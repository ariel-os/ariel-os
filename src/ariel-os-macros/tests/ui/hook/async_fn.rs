#![no_main]

// FAIL: the function must not be async
#[ariel_os::hook(usb_builder)]
async fn main() {}
