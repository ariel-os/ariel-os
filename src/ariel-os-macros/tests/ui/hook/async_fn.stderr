error: custom attribute panicked
 --> tests/ui/hook/async_fn.rs:4:1
  |
4 | #[ariel_os::hook(usb_builder)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: hook functions cannot be async
